src/cmd/builtin/shutdown.c

```diff
@@ -12,13 +12,54 @@
 # include <config.h>
 #endif
 #include <unistd.h>
+#include <jansson.h>
 #include <flux/core.h>
 
 #include "src/broker/state_machine.h"
+#include "src/common/libkvs/kvs_checkpoint.h"
 #include "src/common/libutil/uri.h"
 
 #include "builtin.h"
 
+static void get_kvs_version (flux_t *h, int *version)
+{
+    (*version) = 0;
+    if (flux_kvs_get_version (h, NULL, version) < 0
+        && errno != ENOSYS)
+        log_err_exit ("Error fetching KVS version");
+}
+
+static void get_gc_threshold (flux_t *h, int *gc_threshold)
+{
+    flux_future_t *f;
+    json_t *o;
+    (*gc_threshold) = 0;
+    if (!(f = flux_rpc (h, "config.get", NULL, FLUX_NODEID_ANY, 0))
+        || flux_rpc_get_unpack (f, "o", &o) < 0)
+        log_msg_exit ("Error fetching flux config: %s",
+                      future_strerror (f, errno));
+    (void)json_unpack (o, "{s:{s:i}}", "kvs", "gc-threshold", gc_threshold);
+}
+
+static bool gc_threshold_check (flux_t *h)
+{
+    int gc_threshold, version;
+    bool rc = false;
+
+    get_kvs_version (h, &version);
+    get_gc_threshold (h, &gc_threshold);
+
+    if (gc_threshold > 0 && version > gc_threshold) {
+        char *s = NULL;
+        printf ("gc threshold exceeded, do you want to perform garbage collection (Y/n)? ");
+        scanf ("%ms", &s);
+        if (!s || strncasecmp (s, "y", 1) == 0)
+            rc = true;
```

Comment:
If I just type a carriage return here, I get no feedback but scanf is still waiting for input.  Maybe something like the following would work better, and would handle stdin not a tty also (e.g. when scripted)
```c
bool askyn (char *prompt, bool default_value)
{
    char buf[16];
    int i;

    for (;;) {
        printf ("%s [%s]? ", prompt, default_value ? "Y/n" : "y/N");
        fflush (stdout);
        if (!isatty (STDIN_FILENO)
            || fgets (buf, sizeof (buf), stdin) == NULL
            || buf[0] == '\n')
            break;
        if (buf[0] == 'y' || buf[0] == 'Y')
            return true;
        if (buf[0] == 'n' || buf[0] == 'N')
            return false;
        printf ("Please answer y or n\n");
    };
    return default_value;
}
```

Reply:
looks good, and it seems that several builders didn't like the newer `%ms` that (I thought) was supported in all newer `scanf`s

Reply:
on second thought, it might be a little weird to take the default on stdin EOF or `!isatty()`.  Perhaps this function should be modified so that it can return an error in those cases, and the caller could do something like abort and suggest a `-y` option?

Also, this seems like it could be a useful libutil function for reuse...